### PR TITLE
Add WSL

### PR DESCRIPTION
**Pull request by dsame**
# Description
New tool

Installing Windows Subsystem for Linux is turning on Windows Optional feature. Installation time: 1sec

## Check list
- [x] Related issue / work item is attached
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/Installers/Windows2019/Install-WSL1.ps1 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+################################################################################
+##  File:  Install-WSL1.ps1
+##  Desc:  Install Windows Subsystem for Linux
+################################################################################
+
+Write-Host "Install Windows Subsystem for Linux"
+
+Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Windows2019/Install-WSL1.ps1:8]: Will this override default bash that come from Git?

Comment (dsame) [reply images/win/scripts/Installers/Windows2019/Install-WSL1.ps1]: No, wsl does not include any linux distribution (user should choose one) and even if it did the linux command need to be invoked specially `wsl ...`

File: images/win/scripts/Installers/Validate-WSL.ps1 [hunk 2]
```diff
@@ -0,0 +1,24 @@
+################################################################################
+##  File:  Validate-MongoDB.ps1
+##  Desc:  Validate MongoDB
+################################################################################
+
+if (Get-Command -Name 'mongod')
```
Comment (AlenaSviridenko) [inline images/win/scripts/Installers/Validate-WSL.ps1:6]: This should be WSL, right?

Comment (AlenaSviridenko) [reply images/win/scripts/Installers/Validate-WSL.ps1:6]: You are still using `mongod` in the conditions

File: images/win/Windows2019-Azure.json [hunk 3]
```diff
@@ -110,6 +110,14 @@
             ],
             "execution_policy": "unrestricted"
         },
+        {
+            "type": "powershell",
+            "elevated_user": "SYSTEM",
+            "elevated_password": "",
+            "scripts":[
+                "{{ template_dir }}/scripts/Installers/Windows2019/Install-WSL1.ps1"
```
Comment (maxim-lobanov) [inline images/win/Windows2019-Azure.json:118]: Let's use consistent naming `Install-WSL1`, `Validate-WSL`

File: images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1 [hunk 4]
```diff
@@ -22,6 +22,11 @@ $markdown += New-MDList -Style Unordered -Lines @(
     "Image Version: $env:ImageVersion"
 )
 
+$markdown += New-MDHeader "Enabled windows optional features" -Level 2
+$markdown += New-MDList -Style Unordered -Lines @(
+    "Windows Subsystem for Linux"
```
Comment (maxim-lobanov) [inline images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1:27]: Can we determine and list all enabled features automatically somehow?

Comment (dsame) [reply images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1:27]: Yes, but the list is huge and the titles are vague, this why i prefer to mention only the features we turn on explicitly

**Comment by Manishearth** [comment]
I think our Windows builds started failing because of this, see https://github.com/rust-lang/rust/pull/73566#issuecomment-659545659

**Comment by Manishearth** [comment]
We're fixing it by explicitly setting the shell: https://github.com/rust-lang/rust/pull/74406

**Comment by The-Compiler** [comment]
FWIW if you have some kind of test which relies on `bash` working when it's available (and which gets skipped if not), this change likely will break it. In pytest, we moved to instead checking `bash --version` output: https://github.com/pytest-dev/pytest/pull/7520

**Comment by al-cheb** [comment]
Hello, @The-Compiler.
After installing WSL feature we have got another place which contains `bash.exe - C:\Windows\System32\bash.exe`.  On Windows, the `subprocess` class uses the [Windows CreateProcess() function](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa?redirectedfrom=MSDN). If the file name does not contain a directory path, the system searches for the executable file in the following sequence:

1. The directory from which the application loaded.
2. The current directory for the parent process.
**3. The 32-bit Windows system directory. Use the `GetSystemDirectory` function to get the path of this directory.**
4. The 16-bit Windows system directory. There is no function that obtains the path of this directory, but it is searched. The name of this directory is System.
5. The Windows directory. Use the `GetWindowsDirectory` function to get the path of this directory.
6. The directories that are listed in the PATH environment variable. Note that this function does not search the per-application path specified by the App Paths registry key. To include this per-application path in the search sequence, use the ShellExecute function.

 If the file name does not contain an extension, .exe is appended, therefore bash is equivalent to bash.exe . Using logic above it first looks in the `GetSystemDirectory = C:\Windows\System32` directory ,  before than in the PATH environment variable.

In that case you should use follow approaches:
1. Add the parameter  `shell=True` - https://bugs.python.org/issue8557, https://stackoverflow.com/questions/24789749/popen-with-conflicting-executable-path
`subprocess.run(..., shell=True)`

2. Set  a fully qualified path to a bash shell application:
`subprocess.run(['C:\\Program Files\\Git\\bin\\bash.exe']....)`

**Comment by umarcor** [comment]
@The-Compiler I found the same issue with GitHub Actions and pytest. The following would fail:

```py
    def setUp(self):
        self.shell = ['bash'] if platform == 'win32' else []
        print('\n::group::Log')
        sys.stdout.flush()

    def tearDown(self):
        print('\n::endgroup::')
        sys.stdout.flush()

    def _sh(self, args):
        check_call(self.shell + args, stderr=STDOUT)
```

Instead of setting `shell=True` or hardcoding the full path, I could work around it using `which` from `shutil`:


```py
    def setUp(self):
        self.shell = [which('bash')] if platform == 'win32' else []
        print('\n::group::Log')
        sys.stdout.flush()
```

This seems to look into the PATH, skipping the `GetSystemDirectory` function that @al-cheb commented.

**Comment by H-M-H** [comment]
Another alternative that makes `CreateProcess` work again is to just delete: `C:\Windows\System32\bash.exe`

**Pull request merged**